examples/recipes/msgspec_main.py

```diff
@@ -0,0 +1,95 @@
+from datetime import datetime
+from datetime import timezone
+from functools import partial
+from http import HTTPStatus
+from typing import Annotated, Any
+import uuid
+
+import msgspec
+
+import falcon
+from falcon import Request
+from falcon import Response
+from falcon.media import JSONHandler
+
+
+def _utc_now() -> datetime:
+    return datetime.now(timezone.utc)
+
+
+class Note(msgspec.Struct):
+    text: Annotated[str, msgspec.Meta(max_length=256)]
+    noteid: uuid.UUID = msgspec.field(default_factory=uuid.uuid4)
+    created: datetime = msgspec.field(
+        default_factory=partial(datetime.now, timezone.utc)
+    )
+
+
+class NoteResource:
+    POST_SCHEMA = Note
+
+    def __init__(self) -> None:
+        self._store = {}
+
+    def on_get_note(self, req: Request, resp: Response, noteid: uuid.UUID) -> None:
+        resp.media = self._store.get(str(noteid))
+        if not resp.media:
+            raise falcon.HTTPNotFound(
+                description=f'Note with {noteid=} is not in the store.'
+            )
+
+    def on_delete_note(self, req: Request, resp: Response, noteid: uuid.UUID) -> None:
+        self._store.pop(str(noteid), None)
+        resp.status = HTTPStatus.NO_CONTENT
+
+    def on_get(self, req: Request, resp: Response) -> None:
+        resp.media = self._store
+
+    def on_post(self, req: Request, resp: Response, note: Note) -> None:
+        self._store[str(note.noteid)] = note
+        resp.location = f'req.path/{note.noteid}'
```

Comment:
missing {} in req.path

Reply:
Ouch, nice catch. Will fix and add a proper test.